### PR TITLE
CRM-20032 event config: payment processor selection requires administer CiviCRM permission

### PR DESCRIPTION
**Pull request by francescbassas**
Overview
----------------------------------------
https://issues.civicrm.org/jira/browse/CRM-20032

Before
----------------------------------------
currently, in order to select a payment processor on the event fee tab, you must have the "administer CiviCRM" permission

After
----------------------------------------
in addition to "administer CiviCRM", users with "access CiviEvent" and "Edit All Events" can select payment processor

Technical Details
----------------------------------------
-

Comments
----------------------------------------
-

### CODE REVIEW

File: CRM/Core/Permission.php [hunk 1]
```diff
@@ -1367,6 +1367,14 @@ public static function getEntityActionPermissions() {
         'edit contributions',
       ),
     );
+    
+    // Payment Processor Permissions    
+    $permissions['payment_processor'] = array(
+      'get' => array(
+        array('access CiviEvent', 'edit all events'),
```
Comment (JoeMurray) [inline CRM/Core/Permission.php]: why two arrays?

**Comment by JoeMurray** [comment]
I'm a bit confused: why the focus just on the payment processor field? Is the issue that administer CiviCRM allows editing an event page including that field, but edit all events/access civievent without administer CiviCRM allows editing all but that field? And just to confirm...this is all backend administration that we're talking about, not selecting which processor to pay with on the front end when multiple processors have been enabled for an Event registration page.

File: CRM/Core/Permission.php (hunk 1 above)
Comment (francescbassas) [reply CRM/Core/Permission.php]: I'm trying to grant ("access CiviEvent" AND "edit all events") OR "administer CiviCRM" as commented [here](https://issues.civicrm.org/jira/browse/CRM-20032?focusedCommentId=100914&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-100914) and I understand this is the way as documented [here](https://github.com/francescbassas/civicrm-core/blob/97549a974fbf1b99875bd8ae65bfab394613a802/CRM/Core/Permission.php#L922)

**Comment by francescbassas** [comment]
Sorry, I was checking this bug affecting us on a 4.7.22 but I just realized that something is changed on backend event fee form on a 5.3.1 that has caused it has been fixed unintentionally? I will close this PR for now and will finish doing more tests to ensure that the error no longer exists.

**4.7.22**


**5.3.1**

**Comment by MegaphoneJon** [comment]
IIRC this was actually fixed intentionally, I think by @yashodha?  I recall finding this fixed myself.  And yes @JoeMurray - that's the only field that requires "Administer CiviCRM".  It actually caused anyone without that permission to break an event page they edited before this was fixed.

File: CRM/Core/Permission.php (hunk 1 above)
Comment (JoeMurray) [reply CRM/Core/Permission.php]: Thanks for enlightening me. Looks good.

**Comment by JoeMurray** [comment]
Thanks Jon. It makes me think there was a use case for event organizers to be allowed to edit everything but the money field. I like it simpler without that unusual pattern.

**Comment by yashodha** [comment]
@francescbassas @MegaphoneJon Yes, this has been fixed under https://github.com/civicrm/civicrm-core/pull/11153

**Comment by francescbassas** [comment]
Thanks all! Anyone with permissions can close this? https://issues.civicrm.org/jira/browse/CRM-20032

**Pull request closed without merging**